assets/js/components/FragmentHandler.js

```diff
@@ -15,54 +15,39 @@ module.exports = class FragmentHandler {
     this.window = _window;
     this.doc = doc;
     this.isSingleton = true;
-    this.window.addEventListener('DOMContentLoaded', this.handleDomLoad.bind(this));
+    this.window.addEventListener('readystatechange', this.onReadyStateChange.bind(this));
+    this.onReadyStateChange({});
   }
 
-  handleDomLoad(e) {
-    this.handleSectionOpeningViaHash(e);
-    this.window.addEventListener('hashchange', this.handleSectionOpeningViaHash.bind(this));
-  }
-
-  /**
-   * Indicates whether an HTML element with a given id is found as/within a specific chunk of html
-   *
-   * @param {String} id The id  of the HTML element to search for
-   * @param {HTMLElement} $section The element to search/search within
-   * @param {HTMLDocument} doc
-   * @param {Function} areElementsNested Handles determination of nesting
-   * @returns {boolean} true if element with id is $section, or is contained within $section
-   */
-  isIdOfOrWithinSection(id, $section, doc, areElementsNested) {
-    if (id.search(/https?:\/\//) === 0) {
-      return false;
-    }
-
-    if (id === $section.id) {
-      return true;
+  onReadyStateChange(e) {
+    if (document.readyState !== 'loading') {
+      this.initialise(e);
+      this.window.removeEventListener('readystatechange', this.onReadyStateChange.bind(this));
     }
+  }
 
-    const $fragWithId = doc.querySelector('#' + id);
-    return areElementsNested($section, $fragWithId);
+  initialise(e) {
+    this.handleSectionOpeningViaHash(e);
+    this.window.addEventListener('hashchange', this.handleSectionOpeningViaHash.bind(this));
   }
 
   /**
    * Returns the id of the collapsed section containing the html element with idToFind, or null.
    *
    * @param {String} idToFind The id to search for
    * @param {HTMLDocument} doc
-   * @param {Function} areElementsNested Handles determination of nesting
    * @returns {String} The id of the collapsed section containing idToFind, or null
    */
-  getIdOfCollapsedSection(idToFind, doc, areElementsNested) {
+  getIdOfCollapsedSection(idToFind, doc) {
     let collapsedSections = doc.querySelectorAll('.article-section--collapsed');
-    if (!collapsedSections) {
+    if (!collapsedSections.length) {
       return null;
     }
 
     let $collapsedSectionContainingFrag;
     [].forEach.call(collapsedSections, ($collapsedSection) => {
       if (!$collapsedSectionContainingFrag) {
-        if (this.isIdOfOrWithinSection(idToFind, $collapsedSection, doc, areElementsNested)) {
```

Comment:
perhaps we should move in the future to tell all sections `this fragment has been selected` rather than telling a single one `open`. That would move keep this logic inside `ArticleSection`, but it's out of scope for the bugfix